lib/protobuf/decoder.ex

```diff
@@ -142,32 +142,43 @@ defmodule Protobuf.Decoder do
             type: type,
             oneof: oneof,
             name_atom: name_atom,
-            embedded?: embedded
+            embedded?: embedded,
+            options: options
           } = prop
       } ->
         key = if oneof, do: oneof_field(prop, msg_props), else: name_atom
 
         struct =
-          if embedded do
-            embedded_msg = decode(val, type)
+          if not is_nil(options) do
```

Comment:
To reduce code duplication, it could use:

```elixir
if is_nil(options) or not embedded do
...
else
  embedded_msg = unless is_nil(options), do: Protobuf.FieldOptionsProcessor.decode_type(val, type, options), else: decode(val, type)
  ....
```

Reply:
fixed